orchagent/buffer/buffercontainer.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdbool>
+#include <cstdint>
+
+#include <unordered_map>
+#include <unordered_set>
+#include <string>
+
+class BufferContainer
+{
+public:
+    BufferContainer() = default;
+    virtual ~BufferContainer() = default;
+
+    std::unordered_map<std::string, std::string> fieldValueMap;
+};
+
+class BufferProfileConfig final : public BufferContainer
+{
+public:
+    BufferProfileConfig() = default;
+    ~BufferProfileConfig() = default;
+
+    inline bool isTrimmingProhibited() const
+    {
+        return ((pgRefCount > 0) || (iBufProfListRefCount > 0) || (eBufProfListRefCount)) ? true : false;
```

Comment:
Can we avoid this refcount to determine the trim eligibility?

Reply:
if possible, buffer_profile-->buffer_pool(check ingress/egress)?

Reply:
@stephenxs any other option?

Reply:
> if possible, buffer_profile-->buffer_pool(check ingress/egress)?

@kperumalbfn Buffer Profile to Buffer Pool relation check (`resolveFieldRefArray`) is implemented as part of:
`task_process_status BufferOrch::processPriorityGroup(KeyOpFieldsValuesTuple &tuple)`
`task_process_status BufferOrch::processIngressBufferProfileList(KeyOpFieldsValuesTuple &tuple)`
`task_process_status BufferOrch::processEgressBufferProfileList(KeyOpFieldsValuesTuple &tuple)`

Trimming eligibility validation serves different purposes and takes place long after this check, so it is not relevant